internal/api/v1beta1/servicedata.go

```diff
@@ -87,7 +88,12 @@ func (h Handler) UpsertUserServiceData(ctx context.Context, request *shieldv1bet
 		return nil, grpcBadBodyError
 	}
 
-	if len(requestBody.Data) > h.serviceDataConfig.MaxUpsert {
+	sdMap, err := metadata.Build(request.GetBody().GetData().AsMap())
```

Comment:
Do we really need to use `metadata.Build` for this?

Can't we just convert the data as map?